### PR TITLE
Feature/jul implementation

### PR DESCRIPTION
**Pull request by pmlopes**
This PR implements the contextual logging for JUL applications

### CODE REVIEW

File: src/test/resources/vertx-default-jul-logging.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#  The Eclipse Public License is available at
+#  http://www.eclipse.org/legal/epl-v10.html
+#
+#  The Apache License v2.0 is available at
+#  http://www.opensource.org/licenses/apache2.0.php
+#
+#  You may elect to redistribute this code under either of these licenses.
+#
+.level=INFO
+io.netty.util.internal.PlatformDependent.level=SEVERE
```
Comment (tsegismont) [inline src/test/resources/vertx-default-jul-logging.properties]: Is there a need for this one? In theory, `JULTestLogger` only captures the relevant output.

File: src/main/java/io/github/tsegismont/vertx/contextual/logging/JULContextualDataFormatter.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+ * @author Paulo Lopes
+ */
+public final class JULContextualDataFormatter extends Formatter {
+
+  private static final String placeholderPrefix = "%{";
+  private static final String placeholderSuffix = "}";
+
+  private final String template;
+
+  private final Date dat = new Date();
```
Comment (tsegismont) [inline src/main/java/io/github/tsegismont/vertx/contextual/logging/JULContextualDataFormatter.java:44]: Is there a guarantee that this object won't be changed until the value is read?

File: src/main/java/io/github/tsegismont/vertx/contextual/logging/JULContextualDataFormatter.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+          // need to lock to use inside lambda
+          final String defValue;
+          final String ctxKey;
+
+          if (defIndex != -1) {
+            ctxKey = placeholder.substring(0, defIndex);
+            defValue = placeholder.substring(defIndex + 2);
+          } else {
+            defValue = "-";
```
Comment (tsegismont) [inline src/main/java/io/github/tsegismont/vertx/contextual/logging/JULContextualDataFormatter.java]: The other converters use an empty string when no default is provided. How about doing the same for consistency?

**Review by tsegismont — COMMENTED** [review]
@pmlopes thanks! LGTM, just a few comments

**Comment by pmlopes** [comment]
The date object is updated during the format phase call (this is how it also is implemented on the jdk 8)

I'll create a default constant for empty values and remove the jul properties as they are not used in the test as the custom handler is capturing the output instead.

**Comment by pmlopes** [comment]
@tsegismont all issues addressed.

**Comment by tsegismont** [comment]
@pmlopes looks good to me. I noticed you kept a non-empty default value, unlike the other converters. Is this intentional?

**Comment by tsegismont** [comment]
@pmlopes thank you!

**Pull request merged**